src/Autofac/RegistrationExtensions.Generics.cs

```diff
@@ -25,7 +25,7 @@ public static partial class RegistrationExtensions
         /// <param name="implementer">The open generic implementation type.</param>
         /// <returns>Registration builder allowing the registration to be configured.</returns>
         public static IRegistrationBuilder<object, ReflectionActivatorData, DynamicRegistrationStyle>
-            RegisterGeneric(this ContainerBuilder builder, Type implementer)
+            RegisterGeneric(this ContainerBuilder builder, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type implementer)
```

Comment:
I won't mark all of these, but it'd be interesting to consider how we end up handling custom `IConstructorFinder` implementations that potentially allow for non-public constructor usage. I wonder if we may have to keep _all_ constructors even if the use of non-public constructors is like the 99% case.

Reply:
Or maybe this is a case where we have to mark something around the use of the `IConstructorFinder` with that `[RequiresUnreferencedCode]` attribute to issue a warning. Hmmm.